src/zsh/scripts/eval/set-prompt

```diff
@@ -27,14 +27,26 @@ __am_prompt_emit_prompt_arrow() {
     fi
 }
 
+__am_prompt_show_virtualenv() {
+    if [[ -n "$VIRTUAL_ENV" && -n "$DIRENV_DIR" ]]; then
+        setopt PROMPT_SUBST
+        export DIR_ENV=" ($(basename $VIRTUAL_ENV))"
```

Comment:
avoid the export if possible. AM_PROMPT_ARROW is exported because some folks asked to be able to change the arrow to something they liked, which is supported via edit-vars.

Also, I would try to use a bash built-in to derive the base name, something like:

DIR_ENV=${VIRTUAL_ENV%%/*}

basename forks a process and will be slower to execute.

Reply:
Updated to use a built-in instead of basename.

I moved the function to an eval for bash. Based on the documentation for direnv, if it's not a function it won't be seen be by the sub-bash process.